rdkit/Chem/LigandExtract.py

```diff
@@ -0,0 +1,73 @@
+"""Ligand Extraction from mol object (RDKit PDB file reader should be used in mol object generation)
+Originally written by Greg Landrum and updated by Malitha Humayun Kabir as a part of GSoC 2017 
+Project : RDKit - 3Dmol.js integration
+Mentors: Paul Czodrowski and Greg Landrum
+Date: 28th July 2017
+Email# malitha12345@gmail.com
+"""
+
+from collections import namedtuple
+from rdkit import Chem
+
+## Ligand Extract
+
+ExtractResult = namedtuple('ExtractResult',('match','rest'))
+def ExtractMolAtomsMatchingQuery(mol,func, confId,sanitize, includeAttachedHs):
+    """ func should take an atom index """
+    
+    match = [x for x in range(mol.GetNumAtoms()) if func(x)]
+    
+    if includeAttachedHs:
+        # bring over H atoms attached to the atoms matching the query (if necessary)
+        for aid in match:
+            for nbr in mol.GetAtomWithIdx(aid).GetNeighbors():
+                if nbr.GetAtomicNum()==1 and nbr.GetIdx() not in match:
+                    match.append(nbr.GetIdx())
+                    
+    ##### This chunk clones the input molecule and deletes atoms###
+    res2 = Chem.RWMol(mol)
+    for i in sorted(match, reverse=True):
+        res2.RemoveAtom(i)
+    if sanitize:
+        Chem.SanitizeMol(res2)
+    #### Creating new molecule...
+    res = Chem.RWMol()
+    
+    # start with all atoms and their coordinates:
+    # Should probably also handle multiple conformers
+    oconf = mol.GetConformer(confId)
+    nconf = Chem.Conformer(len(match))
```

Comment:
One of the first languages I've learned was [Fortran](https://en.wikipedia.org/wiki/Fortran). Due to its history of being used with punchcards, people looked for ways to shorten their code and one was the implicit constraint that variable names that start with i, j, k, l, m, n, o are integers. As I 'grew' up with this, I see `nconf` and think of 'number of conformations'. `n` is also often used in maths to indicate numbers of things, so even people younger than me might associate the name with this at first. 

A long speech to say that I would prefer `oldConf` and `newConf` as variable names.